### PR TITLE
dbghlp: Make mutex name unique to the process

### LINKED ISSUE #399: WaitForSingleObjectEx returns WAIT_ABANDONED when collecting backtraces
```
OS Version: windows None (None)
Report Version: 104


Application Specific Information:
assertion failed: `(left == right)`
  left: `128`,
 right: `0`

Thread 0 Crashed:
0   std                             0x7ff7e6d46b7c      std::panicking::begin_panic_fmt (panicking.rs:437)
1   backtrace                       0x7ff7e695cbdc      backtrace::dbghelp::init (dbghelp.rs:315)
2   backtrace                       0x7ff7e695b8f4      backtrace::symbolize::dbghelp::resolve (dbghelp.rs:76)
3   backtrace                       0x7ff7e695995a      [inlined] backtrace::symbolize::resolve_frame_unsynchronized (mod.rs:178)
4   backtrace                       0x7ff7e695995a      backtrace::symbolize::resolve_frame<T> (mod.rs:105)
5   backtrace                       0x7ff7e6958ee8      backtrace::capture::Backtrace::resolve (capture.rs:232)
6   backtrace                       0x7ff7e6958db5      backtrace::capture::Backtrace::new (capture.rs:141)
7   gpu_allocator                   0x7ff7e6875c5f      gpu_allocator::VulkanAllocator::allocate (lib.rs:688)
8   breda                           0x7ff7e67c7a01      breda::renderer::vulkan::allocator::gpu_allocator_wrapper::{{impl}}::alloc (gpu_allocator_wrapper.rs:100)

...
```

https://github.com/rust-lang/backtrace-rs/blob/47069af003d7f4ca62f5b4517a43039c46dfc34d/src/dbghelp.rs#L315

First seen here: https://github.com/Traverse-Research/gpu-allocator/issues/13

We've encountered it twice but haven't been able to trap it in the debugger, I'll update this issue if we have more information around it.

### PR DESCRIPTION
**Pull request by ChrisDenton**
Fixes #399.

### CODE REVIEW

**Comment by ChrisDenton** [comment]
Note: this PR only affects Windows. I'm not sure why the Ubuntu and Android tests are failing.

File: src/dbghelp.rs [hunk 1]
```diff
@@ -277,11 +294,8 @@ pub fn init() -> Result<Init, ()> {
         static LOCK: AtomicUsize = AtomicUsize::new(0);
         let mut lock = LOCK.load(SeqCst);
         if lock == 0 {
-            lock = CreateMutexA(
-                ptr::null_mut(),
-                0,
-                "Local\\RustBacktraceMutex\0".as_ptr() as _,
-            ) as usize;
+            let name = mutex_name();
+            lock = CreateMutexA(ptr::null_mut(), 0, name.as_ptr().cast::<i8>()) as usize;
```
Comment (bjorn3) [inline src/dbghelp.rs:298]: Does windows remove the mutex again once all open handles are closed or will this keep accumulating mutexes until the session ends?

Comment (ChrisDenton) [reply src/dbghelp.rs:298]: https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-createmutexa

> Use the [CloseHandle](https://learn.microsoft.com/en-us/windows/desktop/api/handleapi/nf-handleapi-closehandle) function to close the handle. The system closes the handle automatically when the process terminates. The mutex object is destroyed when its last handle has been closed.

**Comment by p--b** [comment]
Is there anything we can do to help get this PR merged? This bug is biting us in production currently.

**Review by workingjubilee — APPROVED** [review]
Rerunning the tests gets everything but the Android tests passing, and we're not blocking merges on the Android tests, so in this goes.

**Pull request merged**